spirv_common.hpp

```diff
@@ -295,6 +295,20 @@ inline std::string convert_to_string(double t, char locale_radix_point)
 	return buf;
 }
 
+#if defined(__clang__) || defined(__GNUC__)
+#pragma GCC diagnostic pop
+#elif defined(_MSC_VER)
+#pragma warning(pop)
+#endif
+
+class FloatFormatter
+{
+public:
+	virtual ~FloatFormatter() = default;
+	virtual std::string format_float(float value) = 0;
```

Comment:
Is one needed for fp16, or will that hit the format_float part?

Reply:
Currently `float16_t` will be converted to `float` before be converted to string:

https://github.com/KhronosGroup/SPIRV-Cross/blob/42aac916ab5db0cbaf55c2d41f4d063f4ce3955a/spirv_glsl.cpp#L5910-L5913

I think `format_float16` is not needed.